src/devices/honeywell.c

```diff
@@ -0,0 +1,114 @@
+/*
+* *** Honeywell (Ademco) Door/Window Sensors ***
+*
+* Tested with the Honeywell 5811 Wireless Door/Window transmitters
+*
+*/
+
+#include "rtl_433.h"
+#include "pulse_demod.h"
+#include "util.h"
+
+uint16_t crc16_buypass(const uint8_t *data){
+  uint16_t crc = 0xfffe;
+  int i;
+  size_t j;
+
+  for(j=0; j<6; j++){
+    crc ^= ((uint16_t) (data[j])) << 8;
+    
+    for(i=0; i<8; i++){
+      if( (crc & 0x8000) ){
+        crc <<= 1;
+        crc ^= 0x8005;
+      } else
+        crc <<= 1;
+    }
+  }
+
+  return (crc);
+}
+
+static int honeywell_callback(bitbuffer_t *bitbuffer) {
+  if(bitbuffer->num_rows == 0)
+    return 0;
+  
+  int good = 0;
+  char time_str[LOCAL_TIME_BUFLEN];
+        local_time_str(0, time_str);
+
+  const uint8_t *bb;
+  if(bitbuffer->num_rows == 1 && bitbuffer->bits_per_row[0] == 64){
+    good = 1;
+    for(uint16_t i=0; i < 8; i++)
+      bitbuffer->bb[0][i] = ~bitbuffer->bb[0][i];
+
+    bb = bitbuffer->bb[0];
+  }
+
+  if(good){
+    char device_id[6]; 
+    char hex[2];
+    sprintf(hex, "%02x", bb[2]);
+    device_id[0] = hex[1];
+    sprintf(hex, "%02x", bb[3]);
+    device_id[1] = hex[0];
+    device_id[2] = hex[1];
+    sprintf(hex, "%02x", bb[4]);
+    device_id[3] = hex[0];
+    device_id[4] = hex[1];
+    device_id[5] = '\0';
+
+    // Raw Binary
+    char binary [65];
+    for (uint16_t bit = 0; bit < 64; ++bit) {
+        if (bb[bit/8] & (0x80 >> (bit % 8))) {
+                binary[bit] = '1';
+        } else {
+                binary[bit] = '0';
+        }
+    }
+    binary[64] = '\0';
+
+    int heartbeat = 0;
+    uint16_t crc_calculated = crc16_buypass(bb);
+    uint16_t crc = (((uint16_t) bb[6]) << 8) + ((uint16_t) bb[7]);
```

Comment:
crc16_ccitt() was exactly what I needed, thank you!  I've updated the driver file.